web/main/models.py

```diff
@@ -858,19 +849,8 @@ def refresh_search_index(cls):
         with connection.cursor() as cursor:
             try:
                 cursor.execute("REFRESH MATERIALIZED VIEW CONCURRENTLY internal_search_view")
-            except ProgrammingError as e:
-                if e.args[0].startswith('relation "internal_search_view" does not exist'):
-                    cls.create_search_index()
-
-    @classmethod
-    def search(cls, *args, **kwargs):
-        try:
-            return cls._search(*args, **kwargs)
-        except ProgrammingError as e:
-            if e.args[0].startswith('relation "internalsearch_view" does not exist'):
```

Comment:
This doesn't actually work—for one thing the view name is internal_search_view, so the clause is never caught. If that's corrected, it still doesn't work because the database is mid-transaction and can't commit the new indexing step. So I just removed this wrapper and put the "private" method at the top.